Installation issue: adios2@2.11 race condition in cmake

### Steps to reproduce the issue

Issue seen in CI relatively often.

Upstream issue: https://github.com/ornladios/ADIOS2/issues/4850

Upstream upstream issue (cause it's a vendored dep): https://github.com/GTkorvo/ffs/pull/115

### Error message

```
Error copying file "thirdparty/ffs/ffs/cod/pregen_source/Linux/lex.yy.c" to "thirdparty/ffs/ffs".
make[2]: *** [thirdparty/ffs/ffs/CMakeFiles/docs.dir/build.make:85: thirdparty/ffs/ffs/lex.yy.c] Error 1
make[2]: *** Deleting file 'thirdparty/ffs/ffs/lex.yy.c'
make[1]: *** [CMakeFiles/Makefile2:1216: thirdparty/ffs/ffs/CMakeFiles/docs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Information on your system

-

### Additional information

_No response_

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/adios2/package.py
+++ b/repos/spack_repo/builtin/packages/adios2/package.py
@@ -217,6 +217,13 @@ class Adios2(CMakePackage, CudaPackage, ROCmPackage):
     # error: invalid use of incomplete type 'PyFrameObject' {aka 'struct _frame'}
     conflicts("^python@3.11:", when="@:2.7")
 
+    # cmake build race condition
+    patch(
+        "https://github.com/ornladios/ADIOS2/commit/16869cf18cb4bd07d500c3048c3d34d1611674c7.patch?full_index=1",
+        when="@2.11.0",
+        sha256="3af07961975ec6c9023dca182ed19458c021cdf1812d34d9a9e9dad1da60ae75",
+    )
+
     # add missing include <cstdint>
     patch("2.7-fix-missing-cstdint-include.patch", when="@2.7")
 

